packages/builder/src/utils/graphql.ts

```diff
@@ -19,21 +17,21 @@ export enum ChainId {
  */
 const getGraphQLEndpoint = (
   chainId: number,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   reactEnv?: any // ProcessEnv
 ): string | undefined => {
-  const environment = reactEnv || env;
   switch (chainId) {
     case ChainId.MAINNET_CHAIN_ID:
       // eslint-disable-next-line max-len
-      return environment.REACT_APP_SUBGRAPH_URL_MAINNET;
+      return `${process.env.REACT_APP_SUBGRAPH_URL_MAINNET}`;
```

Comment:
This functions returns `string` or `undefined`, now it always returns a string, the function below this one is checking the return value, which now will not work as expected.

Check here: https://github.com/gitcoinco/grants-stack/blob/b338d74a47edf37b9864dccbe8f74393cf2f05d1/packages/builder/src/utils/graphql.ts#LL49C18-L49C18

Reply:
yes the error  is on purpose, why do you want to skip the error? you should have the environment variables defined properly, if you don't get an error here stuff will just not work instead, making things hard to debug

Reply:
but also if you want to remove the error, please remove the error handling logic properly, this will just silence it

Reply:
I'm getting the error for goerli where the environment variable is defined, which is the issue. If it had been for an undefined evv, why would I try to change it.

Reply:
@bhargavaparoksham  if you see the error it means the variable is not set. If you change it to return an empty string it means you are going to have an http request to `""`.

Check the `graphqlFetch` function:

```
  const endpoint = getGraphQLEndpoint(chainId, reactEnv);
  if (endpoint) {
    return fetch(endpoint, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
```

If you `console.log` that env var do what do you have? Which one is it?

Reply:
I just tested in main and if you have `REACT_APP_SUBGRAPH_URL_GOERLI` defined it works fine, otherwise it throws that error, I also tested in staging and it works well, there must be something else going on